lib/protoflow-blocks/src/blocks/sys/write_file.rs

```diff
@@ -60,10 +60,25 @@ impl WriteFile {
 }
 
 impl Block for WriteFile {
-    fn execute(&mut self, _runtime: &dyn BlockRuntime) -> BlockResult {
-        while let Some(_message) = self.input.recv()? {
-            unimplemented!() // TODO
+    fn execute(&mut self, runtime: &dyn BlockRuntime) -> BlockResult {
+        use std::io::prelude::Write;
+
+        runtime.wait_for(&self.path)?;
+
+        let Some(path) = self.path.recv()? else {
+            return Ok(());
+        };
+        let mut file = std::fs::OpenOptions::new()
+            .create(true)
+            .append(true)
+            .open(path)?;
```

Comment:
I'd say it's important that we keep the definitions compatible with fs options. Meaning, that this particular implementation would rather suit `AppendFile` block, for which I'd also have an optional input parameter `create` that is `false`, unless specified otherwise.

This might help reducing the confusion and incorrect usage of such blocks.

Reply:
Generally, I'd agree, but this implementation with the `create|append` flags comes from the [block doc](https://github.com/AsimovPlatform/protoflow/blob/1060183e1d98f11514ae73cd32e344be5f5538b8/lib/protoflow-blocks/src/blocks/sys/write_file.rs#L13):

> A block that writes or appends bytes to the contents of a file.

Beyond that, not sure if we should have the blocks `{ AppendFile, WriteFile }` or something like `{ WriteFile, (TruncateFile + WriteFile) }` to achieve `>>file.txt` and `>file.txt`. I suppose we'll end up having access to most fs operations anyway but will also want some "smart" abstractions.
Could just as well have this block accept `create` and `append` (false implies `truncate`) as parameters, no?